mlflow/store/artifact/databricks_sdk_models_artifact_repo.py

```diff
@@ -0,0 +1,93 @@
+
+def _get_databricks_workspace_client():
+    from databricks.sdk import WorkspaceClient
+
+    return WorkspaceClient()
+
+
+class DatabricksSDKModelsArtifactRepository(CloudArtifactRepository):
+    """
+    Stores and retrieves model artifacts via Databricks SDK, agnostic to the underlying cloud
+    that stores the model artifacts.
+    """
+
+    def __init__(self, model_name, model_version):
+        self.model_name = model_name
+        self.model_version = model_version
+        from mlflow.utils.logging_utils import eprint
+        eprint("Using DatabricksSDKModelsArtifactRepository")
+        model_name_parts = model_name.split(".")
+        catalog = model_name_parts[0]
+        schema = model_name_parts[1]
+        model = model_name_parts[2]
+        self.model_base_path = f"/Models/{catalog}/{schema}/{model}/{model_version}"
+        super().__init__(self.model_base_path)
+
+    def list_artifacts(self, path=None):
+        dest_path = self.model_base_path
+        if path:
+            dest_path = posixpath.join(dest_path, path)
+
+        workspace_client = _get_databricks_workspace_client()
+
+        file_infos = []
+        resp = workspace_client.files.list_directory_contents(dest_path)
+        for directory_entry in resp:
+            relative_path = os.path.relpath(directory_entry.path, self.model_base_path)
+            file_infos.append(
+                FileInfo(
+                    path=relative_path,
+                    is_dir=directory_entry.is_directory,
+                    file_size=directory_entry.file_size,
+                )
+            )
+
+        return sorted(file_infos, key=lambda f: f.path)
+
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_file_path=None):
+        dest_path = self.model_base_path
+        if artifact_file_path:
+            dest_path = posixpath.join(dest_path, artifact_file_path)
+
+        with open(src_file_path, "rb") as f:
+            contents = io.BytesIO(f.read())
+
+        workspace_client = _get_databricks_workspace_client()
+        workspace_client.files.upload(dest_path, contents)
+
+    def log_artifact(self, local_file, artifact_path=None):
+        artifact_file_path = os.path.basename(local_file)
+        if artifact_path:
+            artifact_file_path = posixpath.join(artifact_path, artifact_file_path)
+
+        self._upload_to_cloud(
+            cloud_credential_info=None,
+            src_file_path=local_file,
+            artifact_file_path=artifact_file_path,
+        )
+
+    def _download_from_cloud(self, remote_file_path, local_path):
+        dest_path = self.model_base_path
+        if remote_file_path:
+            dest_path = posixpath.join(dest_path, remote_file_path)
+
+        workspace_client = _get_databricks_workspace_client()
+        resp = workspace_client.files.download(dest_path)
+        contents = resp.contents
+
+        with open(local_path, "wb") as f:
+            f.write(contents.read())
```

Comment:
is there a way to write in chunks like this?

```
for chunk in contents:
    f.write()
```

Reply:
good point. I took a look at databricks-sdk-py, [they're indeed using the `stream` field in the requests call](https://github.com/databricks/databricks-sdk-py/blob/79b096fb414d87ea5c8d4484687544ce63bbe266/databricks/sdk/service/files.py#L833). So we should be able to download the 5GB file in small chunks.
Your comment reminds me that the _upload_to_cloud call is also subject to similar OOM issue(I need to supply a file handle to `upload` instead of reading the file into memory and wrap it as a Bytes IO object)